ui/contexts/gasFee.js

```diff
@@ -0,0 +1,37 @@
+import React, { createContext, useContext } from 'react';
+import PropTypes from 'prop-types';
+import { useGasFeeInputs } from '../hooks/gasFeeInput/useGasFeeInputs';
+
+export const GasFeeContext = createContext({});
+
+export const GasFeeContextProvider = ({
+  children,
+  defaultEstimateToUse,
+  transaction,
+  minimumGasLimit,
+  editGasMode,
+}) => {
+  const gasFeeDetails = useGasFeeInputs(
```

Comment:
Is the benefit of having this here instead of multiple hook invocations simply that there can be no discrepancies? I'm not sure I quite follow what the benefit of this context would be over using the hooks return values directly.

Reply:
Hey @brad-decker : please check these following PRs which use the code from this PR
https://github.com/MetaMask/metamask-extension/pull/12545
https://github.com/MetaMask/metamask-extension/pull/12567

Results from the context can be used in any child component like [this](https://github.com/MetaMask/metamask-extension/pull/12545/files#diff-66d162bb86cf769887e2240412f8950a0f5b3c16f562f75640956f344b03de94R53).

Reply:
@jpuri Brad does raise a question worth discussing and explaining though:

Why is it better to call `useGasFeeContext` on any child component? Instead of just calling `useGasFeeInputs` on any child component?

Reply:
Sure, gas related information are required in a lot of components, we need it for V2 on 3 screens:
1. confirm-transaction
2. edit gas fee
3. advance gas fee

These 3 pages will in-turn have child components thus in all will be about 20 components (approx including containers). If we do not use context we need to pass down the props across all these components. Context eases this props passing in react and is thus useful to keep transient state.

If there is a data field which we want to keep in redux also we can save it in redox from `GasFeeContext` as we do with other react components.

Reply:
Oh, right.

So to further add to the answer to @brad-decker's question is that there is transitional state that is set in `useGasFeeInputs` in response to user action. These are the custom values the user sets for gas fees. We can't just call `useGasFeeInputs` in a bunch of different places, because each `useGasFeeInputs` call is independent of the other. For the `useGasFeeInputs` to have the data we need it to have in all places, we would have to connect it to redux.

This comment from our slack thread explained this further:

> Gotcha. I would say that anything that is part of the "current draft transaction" should be in Redux, in a reducer that represents that draft (e.g. confirm-transaction, though in practice that slice doesn't fully serve that purpose at the moment). But transient data used by gas inputs just while they're being adjusted doesn't belong in Redux, and using component state or React context makes perfect sense for that state (https://consensys.slack.com/archives/GTQAGKY5V/p1635345447101200?thread_ts=1635316808.093100&cid=GTQAGKY5V)